lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+  });
+
+  it('calls onSearch with the current trimmed search text if focus enters the component from elsewhere on the page ' +
+  'while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole} = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox'),
+        anotherElement = document.createElement('button');
+
+    document.body.append(anotherElement);
+    anotherElement.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).toHaveBeenCalledWith('f');
+  });
+
+  it('does not call onSearch if focus moves within the component while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole } = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+
+    const retryBtn = getByRole('button', {name: /retry/i });
+    retryBtn.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+  });
+
+  it('does not call onSearch if focus moves into the component from an outside window while there is an error',
+      function() {
+        const onSearch = jest.fn(),
+            { getByRole } = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+            inputElement = getByRole('combobox');
+
+        expect(onSearch).not.toHaveBeenCalled();
+        inputElement.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+        (document.activeElement as HTMLElement).blur();
+        expect(onSearch).not.toHaveBeenCalled();
+        inputElement.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+      });
+
+  describe('Keyboard Support', function() {
+    it('sets aria-activeddescendant on the combobox and refers to the focused element id' +
+    'when the element has visual focus', async function() {
+      const { getByRole, getAllByRole } = rtlRender({
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+          inputElement = getByRole('combobox');
+
+      inputElement.focus();
+      await userEvent.keyboard('[ArrowDown]');
+      const optionBtns = getAllByRole('option');
+      expect(inputElement).toHaveAttribute('aria-activedescendant', optionBtns[0].id);
+    });
+
+    it('places the editing cursor at the begining of the input field when Home key is pressed', async function() {
+      const onChange = jest.fn(),
+          { getByRole } = rtlRender({ value: 'f', onChange }),
+          inputElement = getByRole('combobox');
+
+      inputElement.focus();
+      await userEvent.keyboard('[Home]o');
+      expect(onChange).toBeCalledWith('of');
+    });
+
+    it('places the editing cursor at the end of the input field when End key is pressed', async function() {
```

Comment:
Probably a good idea to check these both when focus is within the text input and when fake-focus is within the dropdown

Reply:
That is a good idea, here is the update 9439a5d9059e7a2dff2c98fc47e1ac915d559914.